datafusion/expr/src/built_in_function.rs

```diff
@@ -430,7 +430,22 @@ impl BuiltinScalarFunction {
         )
     }
 
-    /// Returns the output [`DataType` of this function
+    /// Returns the dimension [`DataType`] of [`DataType::List`]
```

Comment:
```suggestion
    /// Returns the dimension [`DataType`] of [`DataType::List`].
    /// 
    /// Dimension is defined as the deepest level of nesting.
    /// * `Int64` has dimension 1
    /// * `List(Int64)` has dimension 2
    /// * `List(List(Int64))` has dimension 3
    /// * etc. 
```

Reply:
Proposed PR to add this documentation: https://github.com/apache/arrow-datafusion/pull/7045